Lista2/Ex5/ex5-part1.c

```diff
@@ -0,0 +1,300 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <stdbool.h>
+
+
+
+typedef struct node{
+    int data;
+    struct node *link;
+}NODE;
+
+void clearScreen() { 
+    system("clear"); 
+}
+
+
+
+int emptyList(NODE *head){
+   return (head->link == NULL) ? 1:0;
+}
+
+void startList(NODE *head){
+    head->link = NULL;
+}
+
+void insertionBegin(NODE *head, int x){
+    NODE *newNode =(NODE*)malloc(sizeof(NODE));
+    if(newNode){
+        newNode->data = x;
+        if(emptyList(head)){
+            head->link = newNode;
+            newNode->link = NULL;
+        }
+        else{
+            NODE *oldHead = head->link;
+            head->link = newNode;
+            newNode->link = oldHead;
+        }
+    }
+    else{
+        printf("\n No alocation\n");
+        exit(1);
+    }
+}
+
+void insertionEnd(NODE *head, int x){    
+    NODE  *newNodo = (NODE *)malloc(sizeof(NODE));
+    newNodo->data = x;
+    if(!newNodo){
+        printf("No alocation");
+        exit(1);
+    }
+    else{
+        if(emptyList(head)){ 
```

Comment:
You have the same condition in this function and in the insertionBegin function. Vou falar em portugues o resto pq ainda tenho que terminar de ver o resto. Basicamente, talvez nao fosse viável uma função para ver isso? Ou essa mesma conferencia em algum outro lugar? Só para evitar duplicação de código. Ainda não terminei de ler tudo, nem sei se seria possivel fazer isso, mas foi uma ideia que deu na telha aqui por ele mesmo ja ter falado com a gente.